mpf/modes/game/code/game.py

```diff
@@ -31,16 +32,174 @@ def __init__(self, machine, config, name, path):
         self.num_players = None
         self._stopping_modes = []
         self._stopping_queue = None
+        self._end_ball_event = None  # type: asyncio.Event
+        self._at_least_one_player_event = None  # type: asyncio.Event
 
         self.machine.events.add_handler('mode_{}_stopping'.format(self.name), self._stop_game_modes)
 
+    @asyncio.coroutine
+    def _run(self):
+        """Main game mode method (basic game loop)"""
+
+        # Init the game (game start process)
+
+        # Initialize member variables
+        self.player = None
+        self.player_list = list()
+        self.machine.game = self
+        self.slam_tilted = False
+        self.tilted = False
+        self.num_players = 0
+        self._balls_in_play = 0
+        self._stopping_modes = []
+        self._stopping_queue = None
+        self._end_ball_event = asyncio.Event(loop=self.machine.clock.loop)
+        self._end_ball_event.clear()
+        self._at_least_one_player_event = asyncio.Event(loop=self.machine.clock.loop)
+        self._at_least_one_player_event.clear()
+
+        # Add add player switch handler
+        if self.machine.config['game']['add_player_switch_tag']:
+            self.add_mode_event_handler(
+                self.machine.config['mpf']['switch_tag_event'].replace(
+                    '%', self.machine.config['game']['add_player_switch_tag']),
+                self.request_player_add)
+
+        yield from self._start_game()
+
+        # Game loop
+        while True:
+            # Wait for end ball event to be set
+            yield from self._end_ball_event.wait()
+            if self._end_ball_event.is_set():
```

Comment:
We do not need the if here because the previous line will only return once this is set